python/snewpy/models.py

```diff
@@ -1211,6 +1211,11 @@ def get_initial_spectra(self, t, E, flavors=Flavor):
             Ea = get_value(np.interp(t, self.time, self.meanE[flavor].to('erg')))
             a  = np.interp(t, self.time, self.pinch[flavor])
 
+            # Sanity check to avoid invalid values of Ea, alpha, and L.
+            initialspectra[flavor] = np.zeros_like(E, dtype=float) / (u.erg*u.s)
+            if L <= 0. or Ea <= 0. or a <= -2.:
+                return initialspectra
```

Comment:
This code is inside the `for flavor in flavors` loop, so returning `initialspectra` here could mean we only return values for some flavors, which might crash the calling code, if it expects otherwise.

I think something like
```suggestion
            if L <= 0. or Ea <= 0. or a <= -2.:
                initialspectra[flavor] = np.zeros_like(E, dtype=float) / (u.erg*u.s)
                continue
```
would be better?

Reply:
Quite right, that was a boneheaded mistake. Just committed a fix.